### PR TITLE
Wali/python 3.12 release with automation update

### PR DESCRIPTION
**Pull request by waliMSFT**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as \#\<issue\_number\>.
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

File: src/BuildScriptGenerator/PythonVersions.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public static class PythonVersions
         public const string Python39Version = "3.9.15";
         public const string Python310Version = "3.10.8";
         public const string Python311Version = "3.11.0";
+        public const string Python312Version = "3.12.0";
```
Comment (william-msft) [inline src/BuildScriptGenerator/PythonVersions.cs:18]: Could we add tests for 3.12? You could reference what we're doing for `Python311Version`.

Comment (waliMSFT) [reply src/BuildScriptGenerator/PythonVersions.cs:18]: yah we should. I will add the test. Thanks.

File: build/constants.yaml [hunk 2]
```diff
@@ -148,7 +148,7 @@
       file-name-prefix: __
 - name: python-versions
   constants:
-    python-runtime-base-tag: 20230901.1
+    python-runtime-base-tag: 20231024.1
```
[the remark this answers is not in the document]
Comment (waliMSFT) [reply build/constants.yaml]: yah, I have updated the baseimage tag. pushed the changes already.

File: platforms/python/prereqs/build.sh [hunk 3]
```diff
@@ -93,8 +93,13 @@ make -j $(nproc)
 
 make install
 
-if  [ "${PYTHON_VERSION[0]}" == "3" ] && [ "${PYTHON_VERSION[1]}" -ge "10" ]
+# python3.12.0 version requires the updated pip version. The older pip version:21.2.4 does not work with python3.12.0.
+IFS='.' read -ra SPLIT_VERSION <<< "$PYTHON_VERSION"
+if [ "${SPLIT_VERSION[0]}" == "3" ] && [ "${SPLIT_VERSION[1]}" -ge "12" ]
 then
+    export PATH=/opt/python/$PYTHON_VERSION/bin:$PATH
```
[the remark this answers is not in the document]
Comment (waliMSFT) [reply platforms/python/prereqs/build.sh:100]: yah, as we are using the full path we don't need to update the `PATH` env variable. I will delete it.

**Pull request closed without merging**